Libraries/xcsdk/Headers/xcsdk/SDK/Manager.h

```diff
@@ -61,14 +62,18 @@ class Manager {
     /*
      * Find an SDK by name. This does a fuzzy search, so the "name" could
      * be an SDK name or path, or even the name of a platform.
+     * The provided filesystem will attempt to resolve the name if the name is
+     * a symlinked path.
      */
-    Target::shared_ptr findTarget(std::string const &name) const;
+    Target::shared_ptr findTarget(libutil::Filesystem const *filesystem, std::string const &name) const;
```

Comment:
Why is the filesystem optional?

Reply:
Ah, you answered that in your commit message:

>  Most locations have been stubbed with a
> nullptr for the FS since those are internally-provided toolchain names
> / paths, but some locations that can take in user-provided input are
> different.

I don't understand this, but it sounds like you've thought about it already and I don't have a good reason for the filesystem to be required.

Reply:
Ultimately, I'm just using the filesystem to resolve symlinks.  If no filesystem, no symlink resolution, but for internal invocations that's OK since xcbuild won't call this internally with symlinks.  Only user-facing tools could possibly pass a symlink (from what I've read of the code, anyway).